diff --git a/package.json b/package.json
index bfb23d4..d1f2fa3 100644
--- a/package.json
+++ b/package.json
@@ -221,4 +221,4 @@
     "react-dom": "^18.0"
   },
   "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
-}
+}
\ No newline at end of file
diff --git a/src/components/ContextMenu/__stories__/ContextMenu.stories.tsx b/src/components/ContextMenu/__stories__/ContextMenu.stories.tsx
index d0f5792..f54d782 100644
--- a/src/components/ContextMenu/__stories__/ContextMenu.stories.tsx
+++ b/src/components/ContextMenu/__stories__/ContextMenu.stories.tsx
@@ -206,158 +206,176 @@ const renderSubSelectMenu = (
   };
 
   return (
-    <ContextMenu.Root {...menuProps} key={key}>
-      <ContextMenu.Trigger>
-        <ContextMenuTriggerIcon />
-      </ContextMenu.Trigger>
-
-      <ContextMenu.Portal>
-        <ContextMenu.Content
-          collisionBoundary={
-            document.querySelector('.docs-story') as HTMLElement
-          }
-          {...contentProps}
-        >
-          {additionalLabel && (
-            <ContextMenu.Label
-              text={
-                <Text theme={TextContextMenuTheme} size="l">
-                  {i18n.t('Label')}
-                </Text>
-              }
-            />
-          )}
-
-          <ContextMenu.MetaItem label={i18n.t('Workspace')} value="Kommo" />
-
-          <ContextMenu.MetaItem
-            label={i18n.t('Workspace ID')}
-            value="33764107"
-            isCopyable
-          />
-
-          <ContextMenu.Item
-            text={
-              <Text theme={TextContextMenuTheme} size="l">
-                {i18n.t('Change Workspace')}
-              </Text>
+    <div style={{ position: 'absolute' }}>
+      <div
+        style={{
+          position: 'fixed',
+          zIndex: 700,
+          left: '50%',
+          top: '0',
+          right: '0',
+          bottom: '0',
+          backgroundColor: 'rgba(0,0,0)',
+        }}
+      ></div>
+
+      <ContextMenu.Root {...menuProps} key={key}>
+        <ContextMenu.Trigger asChild>
+          <li className="123">
+            <ContextMenuTriggerIcon />
+          </li>
+        </ContextMenu.Trigger>
+
+        <ContextMenu.Portal>
+          <ContextMenu.Content
+            collisionBoundary={
+              document.querySelector('.docs-story') as HTMLElement
             }
+            {...contentProps}
           >
-            <ContextMenu.ItemRightSlot>
-              <Text theme={TextContextMenuTheme} size="l">
-                ⌘+W
-              </Text>
-            </ContextMenu.ItemRightSlot>
-          </ContextMenu.Item>
-
-          <ContextMenu.Sub>
-            <ContextMenu.SubTrigger
-              text={
-                <Text theme={TextContextMenuTheme} size="l">
-                  {i18n.t('Contacts')}
-                </Text>
-              }
-            />
-
-            <ContextMenu.Portal>
-              <ContextMenu.SubContent>
-                <ContextMenu.Item
-                  text={
-                    <Text theme={TextContextMenuTheme} size="l">
-                      {i18n.t('Contact')} 1
-                    </Text>
-                  }
-                />
-
-                <ContextMenu.Item
-                  text={
-                    <Text theme={TextContextMenuTheme} size="l">
-                      {i18n.t('Contact')} 2
-                    </Text>
-                  }
-                  isDisabled
-                />
-
-                <ContextMenu.Item
-                  icon={<ContextMenuTrashcanIcon />}
-                  text={
-                    <Text theme={TextContextMenuTheme} size="l">
-                      {i18n.t('Delete All')}
-                    </Text>
-                  }
-                  isDanger
-                />
-
-                <ContextMenu.Separator />
-
-                <ContextMenu.Item
-                  text={
-                    <Text theme={TextContextMenuTheme} size="l">
-                      {i18n.t('Technical support')}
-                    </Text>
-                  }
-                />
-              </ContextMenu.SubContent>
-            </ContextMenu.Portal>
-          </ContextMenu.Sub>
-
-          {!menuProps.isMinimalistic && (
-            <>
-              <ContextMenu.Separator />
-
-              <ContextMenu.CheckboxItem
-                isChecked={autoupdateChecked}
-                onChange={handleAutoupdateChange}
+            {additionalLabel && (
+              <ContextMenu.Label
                 text={
                   <Text theme={TextContextMenuTheme} size="l">
-                    {i18n.t('Autoupdate')}
+                    {i18n.t('Label')}
                   </Text>
                 }
-              >
-                <ContextMenu.ItemIndicator>
-                  <ContextMenuCheckIcon />
-                </ContextMenu.ItemIndicator>
-              </ContextMenu.CheckboxItem>
+              />
+            )}
 
-              <ContextMenu.Separator />
+            <ContextMenu.MetaItem label={i18n.t('Workspace')} value="Kommo" />
 
-              <ContextMenu.Label
+            <ContextMenu.MetaItem
+              label={i18n.t('Workspace ID')}
+              value="33764107"
+              isCopyable
+            />
+
+            <ContextMenu.Item
+              text={
+                <Text theme={TextContextMenuTheme} size="l">
+                  {i18n.t('Change Workspace')}
+                </Text>
+              }
+            >
+              <ContextMenu.ItemRightSlot>
+                <Text theme={TextContextMenuTheme} size="l">
+                  ⌘+W
+                </Text>
+              </ContextMenu.ItemRightSlot>
+            </ContextMenu.Item>
+
+            <ContextMenu.Sub>
+              <ContextMenu.SubTrigger
                 text={
                   <Text theme={TextContextMenuTheme} size="l">
-                    {i18n.t('Select Theme')}
+                    {i18n.t('Contacts')}
                   </Text>
                 }
               />
 
-              <ContextMenu.RadioGroup
-                value={theme}
-                onChange={handleThemeChange}
-              >
-                <ContextMenu.RadioItem
-                  value="light"
+              <ContextMenu.Portal>
+                <ContextMenu.SubContent>
+                  <ContextMenu.Item
+                    text={
+                      <Text theme={TextContextMenuTheme} size="l">
+                        {i18n.t('Contact')} 1
+                      </Text>
+                    }
+                  />
+
+                  <ContextMenu.Item
+                    text={
+                      <Text theme={TextContextMenuTheme} size="l">
+                        {i18n.t('Contact')} 2
+                      </Text>
+                    }
+                    isDisabled
+                  />
+
+                  <ContextMenu.Item
+                    icon={<ContextMenuTrashcanIcon />}
+                    text={
+                      <Text theme={TextContextMenuTheme} size="l">
+                        {i18n.t('Delete All')}
+                      </Text>
+                    }
+                    isDanger
+                  />
+
+                  <ContextMenu.Separator />
+
+                  <ContextMenu.Item
+                    text={
+                      <Text theme={TextContextMenuTheme} size="l">
+                        {i18n.t('Technical support')}
+                      </Text>
+                    }
+                  />
+                </ContextMenu.SubContent>
+              </ContextMenu.Portal>
+            </ContextMenu.Sub>
+
+            {!menuProps.isMinimalistic && (
+              <>
+                <ContextMenu.Separator />
+
+                <ContextMenu.CheckboxItem
+                  isChecked={autoupdateChecked}
+                  onChange={handleAutoupdateChange}
                   text={
                     <Text theme={TextContextMenuTheme} size="l">
-                      {i18n.t('Light')}
+                      {i18n.t('Autoupdate')}
                     </Text>
                   }
-                />
+                >
+                  <ContextMenu.ItemIndicator>
+                    <ContextMenuCheckIcon />
+                  </ContextMenu.ItemIndicator>
+                </ContextMenu.CheckboxItem>
+
+                <ContextMenu.Separator />
 
-                <ContextMenu.RadioItem
-                  value="dark"
+                <ContextMenu.Label
                   text={
                     <Text theme={TextContextMenuTheme} size="l">
-                      {i18n.t('Dark')}
+                      {i18n.t('Select Theme')}
                     </Text>
                   }
                 />
-              </ContextMenu.RadioGroup>
-            </>
-          )}
 
-          <ContextMenu.Arrow />
-        </ContextMenu.Content>
-      </ContextMenu.Portal>
-    </ContextMenu.Root>
+                <ContextMenu.RadioGroup
+                  value={theme}
+                  onChange={handleThemeChange}
+                >
+                  <ContextMenu.RadioItem
+                    value="light"
+                    text={
+                      <Text theme={TextContextMenuTheme} size="l">
+                        {i18n.t('Light')}
+                      </Text>
+                    }
+                  />
+
+                  <ContextMenu.RadioItem
+                    value="dark"
+                    text={
+                      <Text theme={TextContextMenuTheme} size="l">
+                        {i18n.t('Dark')}
+                      </Text>
+                    }
+                  />
+                </ContextMenu.RadioGroup>
+              </>
+            )}
+
+            <ContextMenu.Arrow />
+          </ContextMenu.Content>
+        </ContextMenu.Portal>
+      </ContextMenu.Root>
+
+      {contentProps?.direction}
+    </div>
   );
 };
 
@@ -404,14 +422,14 @@ export const Default: Story = {
 
 export const Positions: Story = {
   render: () => {
-    const positions = [
-      { justify: 'flex-start', align: 'flex-start' },
-      { justify: 'center', align: 'flex-start' },
-      { justify: 'flex-end', align: 'flex-start' },
-      { justify: 'flex-start', align: 'flex-end' },
-      { justify: 'center', align: 'flex-end' },
-      { justify: 'flex-end', align: 'flex-end' },
-    ];
+    // const positions = [
+    //   { justify: 'flex-start', align: 'flex-start' },
+    //   { justify: 'center', align: 'flex-start' },
+    //   { justify: 'flex-end', align: 'flex-start' },
+    //   { justify: 'flex-start', align: 'flex-end' },
+    //   { justify: 'center', align: 'flex-end' },
+    //   { justify: 'flex-end', align: 'flex-end' },
+    // ];
 
     const centralPositions = [
       Direction.LEFT_DOWN,
@@ -431,7 +449,7 @@ export const Positions: Story = {
           boxSizing: 'border-box',
         }}
       >
-        {positions.map((pos, i) => (
+        {/* {positions.map((pos, i) => (
           <div
             key={i}
             style={{
@@ -446,7 +464,7 @@ export const Positions: Story = {
               mode: ContextMenuMode.CLICK,
             })}
           </div>
-        ))}
+        ))} */}
 
         <div
           style={{
diff --git a/src/components/ContextMenu/__stories__/Themes.mdx b/src/components/ContextMenu/__stories__/Themes.mdx
index 0dc5fe2..d8cf65a 100644
--- a/src/components/ContextMenu/__stories__/Themes.mdx
+++ b/src/components/ContextMenu/__stories__/Themes.mdx
@@ -8,7 +8,7 @@ import { i18n } from '@i18n';
 
 <Title>{i18n.t('Themes')}</Title>
 
-<Canvas of={ContextMenu.Default} />
+{/* <Canvas of={ContextMenu.Default} /> */}
 
 ## TextContextMenuTheme
 
diff --git a/src/components/ContextMenu/components/Content/Content.module.css b/src/components/ContextMenu/components/Content/Content.module.css
index 86bfa89..0c52f3b 100644
--- a/src/components/ContextMenu/components/Content/Content.module.css
+++ b/src/components/ContextMenu/components/Content/Content.module.css
@@ -4,6 +4,7 @@
   border: 1px solid var(--crm-ui-kit-palette-border-default);
   border-radius: 3px;
   overflow: hidden;
-  box-shadow: 0px 4px 6px -2px rgba(16, 24, 40, 0.03),
+  box-shadow:
+    0px 4px 6px -2px rgba(16, 24, 40, 0.03),
     0px 12px 16px -4px rgba(16, 24, 40, 0.08);
 }
diff --git a/src/components/ContextMenu/components/Content/Content.tsx b/src/components/ContextMenu/components/Content/Content.tsx
index 65d0418..0f7b5eb 100644
--- a/src/components/ContextMenu/components/Content/Content.tsx
+++ b/src/components/ContextMenu/components/Content/Content.tsx
@@ -50,6 +50,7 @@ export const Content = forwardRef<HTMLDivElement, ContentProps>(
   ) => {
     const [hasItemWithIcon, setHasItemWithIcon] = useState(false);
     const [isPositioned, setIsPositioned] = useState(false);
+
     const [align, setAlign] = useState<'start' | 'end'>(
       direction === Direction.UP_RIGHT ||
         direction === Direction.DOWN_RIGHT ||
@@ -258,7 +259,14 @@ export const Content = forwardRef<HTMLDivElement, ContentProps>(
 
     return (
       <LevelProvider hasItemWithIcon={hasItemWithIcon}>
-        <animated.div style={springStyles} data-content-wrapper>
+        <animated.div
+          style={{
+            zIndex: Number.MAX_SAFE_INTEGER,
+            position: 'fixed',
+            ...springStyles,
+          }}
+          data-content-wrapper
+        >
           <RadixDropdownMenuContent
             ref={mergeRefs(contentRef, ref)}
             className={cx(s.content, className)}
diff --git a/src/components/ContextMenu/components/SubContent/SubContent.tsx b/src/components/ContextMenu/components/SubContent/SubContent.tsx
index d8efa8a..b50de6b 100644
--- a/src/components/ContextMenu/components/SubContent/SubContent.tsx
+++ b/src/components/ContextMenu/components/SubContent/SubContent.tsx
@@ -65,7 +65,14 @@ export const SubContent = forwardRef<HTMLDivElement, SubContentProps>(
 
     return (
       <LevelProvider hasItemWithIcon={hasItemWithIcon}>
-        <animated.div style={springStyles} data-content-wrapper>
+        <animated.div
+          style={{
+            position: 'fixed',
+            zIndex: Number.MAX_SAFE_INTEGER,
+            ...springStyles,
+          }}
+          data-content-wrapper
+        >
           <RadixDropdownMenuSubContent
             ref={ref}
             className={cx(s.sub_content, className)}
diff --git a/src/components/ContextMenu/components/Trigger/Trigger.props.ts b/src/components/ContextMenu/components/Trigger/Trigger.props.ts
index 0af3d1d..76e8f82 100644
--- a/src/components/ContextMenu/components/Trigger/Trigger.props.ts
+++ b/src/components/ContextMenu/components/Trigger/Trigger.props.ts
@@ -1,7 +1,6 @@
 import { ComponentPropsWithoutRef } from 'react';
 import { Trigger as RadixDropdownMenuTrigger } from '@radix-ui/react-dropdown-menu';
 
-export type TriggerProps = Omit<
-  ComponentPropsWithoutRef<typeof RadixDropdownMenuTrigger>,
-  'asChild'
+export type TriggerProps = ComponentPropsWithoutRef<
+  typeof RadixDropdownMenuTrigger
 >;
diff --git a/src/components/ContextMenu/components/Trigger/Trigger.tsx b/src/components/ContextMenu/components/Trigger/Trigger.tsx
index bfadacf..06a0b37 100644
--- a/src/components/ContextMenu/components/Trigger/Trigger.tsx
+++ b/src/components/ContextMenu/components/Trigger/Trigger.tsx
@@ -20,9 +20,8 @@ export const Trigger = forwardRef<HTMLButtonElement, TriggerProps>(
 
     return (
       <RadixDropdownMenuTrigger
-        ref={ref}
+        ref={mergeRefs(triggerRef, ref)}
         className={cx(s.button, className)}
-        asChild
         onPointerDown={(e) => {
           if (mode === ContextMenuMode.HOVER) {
             e.preventDefault();
@@ -31,7 +30,7 @@ export const Trigger = forwardRef<HTMLButtonElement, TriggerProps>(
         }}
         {...rest}
       >
-        <button ref={mergeRefs(triggerRef, ref)}>{children}</button>
+        {children}
       </RadixDropdownMenuTrigger>
     );
   }
diff --git a/src/components/ContextMenu/index.ts b/src/components/ContextMenu/index.ts
index f1a8795..d5e59c3 100644
--- a/src/components/ContextMenu/index.ts
+++ b/src/components/ContextMenu/index.ts
@@ -2,5 +2,6 @@ export { ContextMenu } from './ContextMenu';
 export { type ContextMenuRootProps } from './ContextMenu.props';
 export { type ContextMenuType } from './ContextMenu.types';
 export { ContextMenuMode } from './ContextMenu.enums';
+export { Direction as ContextMenuDirection } from './components/Content';
 
 export { TextContextMenuTheme } from './themes';
